client/routes/execution/index.vue

```diff
@@ -116,56 +117,101 @@ export default {
+            return null;
+          }
+
+          if (res.nextPageToken && this.npt === res.nextPageToken) {
+            // nothing happened, and same query is still valid, so let's long pool again
+            return this.fetch(pagedQueryUrl);
           }
-        } else {
-          this.isWorkflowRunning = false;
-        }
 
-        var shouldHighlightEventId = this.$route.query.eventId && this.results.length <= this.$route.query.eventId;
+          if (res.nextPageToken) {
+            this.isWorkflowRunning = JSON.parse(
+              atob(res.nextPageToken)
+            ).IsWorkflowRunning;
+            setTimeout(() => {
+              this.nextPageToken = res.nextPageToken;
+            });
+          } else {
+            this.isWorkflowRunning = false;
+          }
+
+          const shouldHighlightEventId =
+            this.$route.query.eventId &&
+            this.events.length <= this.$route.query.eventId;
+
+          const {events} = res.history;
+          this.events = this.events.concat(events);
 
-        const events = res.history.events;
-        this.events = this.events.concat(events);
+          this.history.events = getHistoryEvents(this.events);
+          this.history.timelineEvents = getHistoryTimelineEvents(
+            this.history.events
+          );
 
-        this.history.events = getHistoryEvents(this.events);
-        this.history.timelineEvents = getHistoryTimelineEvents(this.history.events);
+          this.summary = getSummary({
+            events: this.events,
+            isWorkflowRunning: this.isWorkflowRunning,
+            workflow: this.workflow,
+          });
 
-        this.summary = getSummary({
-          events: this.events,
-          isWorkflowRunning: this.isWorkflowRunning,
-          workflow: this.workflow,
+          if (shouldHighlightEventId) {
+            this.$emit('highlight-event-id', this.$route.query.eventId);
+          }
+
+          return this.events;
+        })
+        .catch(e => {
+          // eslint-disable-next-line no-console
+          console.error(e);
+          // eslint-disable-next-line no-underscore-dangle
+          if (this._isDestroyed || this.pqu !== pagedQueryUrl) {
+            return;
+          }
+          this.history.error =
+            (e.json && e.json.message) || e.status || e.message;
+        })
+        .finally(() => {
+          // eslint-disable-next-line no-underscore-dangle
+          if (this._isDestroyed || this.pqu !== pagedQueryUrl) {
+            this.history.loading = false;
+          }
         });
+    },
+    onBaseApiUrlChange(baseAPIURL) {
+      this.events = [];
+      this.history.events = [];
+      this.history.timelineEvents = [];
+      this.nextPageToken = undefined;
+      return this.$http(baseAPIURL)
+        .then(
+          wf => {
+            this.workflow = wf;
+            this.isWorkflowRunning = !wf.workflowExecutionInfo.closeTime;
+            this.setupQueryUrlWatch();
```

Comment:
This change will setup watcher once first API has been successful, rather than executing both APIs simultaneously.